### PR TITLE
Upgrade to smallrye-config 1.3.9

### PR DESCRIPTION
**Pull request by dmlloyd**

### CODE REVIEW

**Comment by gsmet** [comment]
@dmlloyd coud you clarify why it's a Draft PR? You have identified some issues or you're waiting for something else?

**Comment by dmlloyd** [comment]
Right now it's a draft because I wasn't sure it was going to work, and it didn't ;).  But I think we can make it a "real" PR as soon as I've fixed the problems, which I think I'm quite close to doing.

**Comment by dmlloyd** [comment]
The TCK problem is at the heart of eclipse/microprofile-config#407.  If we decide to go ahead with this, we may have to disable some TCK tests.

**Comment by dmlloyd** [comment]
Windows failure seems unrelated.

File: core/runtime/src/main/java/io/quarkus/runtime/configuration/ConfigInstantiator.java [hunk 1]
```diff
@@ -55,33 +57,47 @@ private static void handleObject(String prefix, Object o, SmallRyeConfig config)
...
+                    } else {
+                        val = config.getOptionalValue(fullName, fieldClass);
+                    }
                     if (val.isPresent()) {
-                        field.set(o, val.get());
+                        field.set(o, fieldIsOptional ? val : val.get());
                     } else if (defaultValue != null) {
-                        if (field.getType().equals(List.class)) {
-                            Class<?> listType = (Class<?>) ((ParameterizedType) field.getGenericType())
+                        if (fieldClass.equals(List.class)) {
```
Comment (gsmet) [inline core/runtime/src/main/java/io/quarkus/runtime/configuration/ConfigInstantiator.java]: You could reuse `fieldIsList` here, couldn't you?

File: tcks/microprofile-config/tck-suite.xml [hunk 2]
```diff
@@ -24,13 +24,23 @@
...
-                        expected value. expected [clean test] but found [ clean test] -->
-                    <exclude name="testEnvironmentConfigSource" />
                     <!-- expected [clean test] but found [ clean test] -->
                     <exclude name="testPropertyConfigSource" />
                 </methods>
             </class>
+            <!-- TCK and spec dispute: https://github.com/eclipse/microprofile-config/pull/407 -->
+            <class name="org.eclipse.microprofile.config.tck.ConfigProviderTest">
+                <methods>
+                    <exclude name="testEnvironmentConfigSource"/>
```
Comment (gsmet) [inline tcks/microprofile-config/tck-suite.xml]: Could you maybe copy the comment you removed above? It would be useful IMHO.

**Review by gsmet — CHANGES_REQUESTED** [review]
I added two very minor comments. Otherwise it looks good.

**Review by gsmet — APPROVED** [review]

**Pull request merged**